lib/pages/chat_details/chat_details_page_view/links/chat_details_links_item.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/preview_url/get_preview_url_success.dart';
+import 'package:fluffychat/pages/chat_details/chat_details_page_view/links/chat_details_links_style.dart';
+import 'package:fluffychat/utils/string_extension.dart';
+import 'package:fluffychat/utils/url_launcher.dart';
+import 'package:fluffychat/widgets/mixins/get_preview_url_mixin.dart';
+import 'package:fluffychat/widgets/mxc_image.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+class ChatDetailsLinkItem extends StatefulWidget {
+  const ChatDetailsLinkItem({
+    super.key,
+    required this.event,
+  });
+
+  final Event event;
+
+  @override
+  State<ChatDetailsLinkItem> createState() => _ChatDetailsLinkItemState();
+}
+
+class _ChatDetailsLinkItemState extends State<ChatDetailsLinkItem>
+    with GetPreviewUrlMixin {
+  String get link {
+    final body = widget.event.body;
+    return body.getFirstValidUrl() ?? '';
+  }
+
+  Uri get uri {
+    return Uri.tryParse(link) ?? Uri();
+  }
+
+  @override
+  String debugLabel = 'ChatDetailsLinkItem';
+
+  @override
+  void initState() {
+    super.initState();
+    getPreviewUrl(uri: uri);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: getPreviewUrlStateNotifier,
+      builder: (context, previewUrlState, child) {
```

Comment:
make use of child for increase performance

Reply:
can't do that, it need previewUrlState

Reply:
yes, but must be have a widget that don't depend on the previewUrlState and can be put in child

Reply:
`_AvatarPlaceholder` i think

Reply:
done